frontend/src/components/ShootDns/DnsProviderRow.vue

```diff
@@ -261,6 +270,9 @@ export default {
       return getValidationErrors(this, field)
     },
     onDelete () {
+      if (this.readonly) {
+        return
+      }
```

Comment:
is this check really needed as the button is anyhow disabled
and if it is needed, why not check for `readonly || primaryReadonly`?

Reply:
No it is not necessary. Yes it should be `this.readonly || this.primaryReadonly`. I have added it because I added the same to the `setData` method. Because I don't want to update the valid state if it is readonly. But we could move this to `updateValid` and also remove it from `setData`.